library/src/scala/quoted/matching/Value.scala

```diff
@@ -0,0 +1,19 @@
+package scala.quoted
+package matching
+
+/** Value expressions */
+object Value {
+
+  /** Matches expressions containing values and extracts the value.
+   *
+   *  Usage:
+   *  ```
+   *  (x: Expr[B]) match {
+   *    case Value(value) => ... // value: B
+   *  }
+   *  ```
+   */
+  def unapply[T](expr: Expr[T])(given valueOf: ValueOfExpr[T], qxtc: QuoteContext): Option[T] =
+    valueOf(expr)
+
+}
```

Comment:
We already have extractor for Const, a duplicate? The name `Value` is a little misleading, if we in fact only want to get the literal const out.

Reply:
Yes, Value is more general. I plan to replace and remove Const in another PR